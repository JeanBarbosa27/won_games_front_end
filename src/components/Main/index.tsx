import * as S from './styles'

const Main = ({
  title = 'React Avançado - Boilerplate',
  description = 'TypeScript, ReactJS, NextJS e Styeld Components!',
  background = ''
}) => (
  <S.Wrapper style={{ backgroundColor: background }}>
    <S.Logo
      src="/img/logo.svg"
      alt="Image de um átomo e React Avançado ao lado."
    />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="Pessoa desenvolvendo código de programação no computador"
    />
  </S.Wrapper>
)

export default Main
