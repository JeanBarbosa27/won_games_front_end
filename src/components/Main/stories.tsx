import { Story, Meta } from '@storybook/react/types-6-0'

import Main from '.'

export default {
  title: 'Main',
  component: Main,
  argTypes: {
    background: {
      control: 'color'
    }
  },
  args: {
    title: 'Title Default',
    description: 'Description Default'
  }
} as Meta

export const Basic: Story = (args) => <Main {...args} />
Basic.args = {
  title: "Basic's title",
  description: "Description's title",
  background: '#012787'
}

export const Default: Story = (args) => <Main {...args} />
