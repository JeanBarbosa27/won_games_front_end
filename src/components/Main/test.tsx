import { render, screen } from '@testing-library/react'

import Main from '.'
// import { colors } from 'styles/variables'

describe('<Main />', () => {
  it('should be able to render Main componente', () => {
    const { container } = render(<Main />)
    const heading = screen.getByRole('heading', { name: /react avanÃ§ado/i })

    expect(heading).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyleRule('background-color', '#06092b')
  })
})
