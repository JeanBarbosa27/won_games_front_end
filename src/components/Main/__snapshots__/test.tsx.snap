// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should be able to render Main componente 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  padding: 3rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  color: #ffffff;
  background-color: #06092b;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Image de um átomo e React Avançado ao lado."
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    React Avançado - Boilerplate
  </h1>
  <h2
    class="c3"
  >
    TypeScript, ReactJS, NextJS e Styeld Components!
  </h2>
  <img
    alt="Pessoa desenvolvendo código de programação no computador"
    class="c4"
    src="/img/hero-illustration.svg"
  />
</main>
`
